pkg/cmd/server/origin/run_components.go

```diff
@@ -265,9 +268,15 @@ func (c *MasterConfig) RunDeploymentController() {
 	if err != nil {
 		glog.Fatalf("Unable to initialize deployment controller: %v", err)
 	}
+
+	port, err := kubeserver.GetKubernetesMasterPort(&c.Options)
+	if err != nil {
+		glog.Fatal(err)
+	}
+	master := fmt.Sprintf("https://kubernetes.default.svc.%s:%d", c.Options.DNSConfig.Domain, port)
```

Comment:
@liggitt The env vars {KUBERNETES,OPENSHIFT}_MASTER must be match a alternative name in the x509 certificate. My assumption was that one would sign the fully qualified domain name only. The k8s install does sign the full DNS chain.


Example from one of my test clusters:
```
                IP Address:172.17.42.1, IP Address:192.168.0.5, IP Address:10.254.0.1, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.kubernetes.io, DNS:roque-kube-master
```

It is probably preferable to copy the approach used upstream and have the same alternate names... the possibility of a kubernetes master impersonating for another one because one uses the non-fully qualified method is probably not relevant.

Regarding the port. The port used by the kubernetes master is the one that should be used for this env var. AFAIK, i'm using the same port as is being passed as the service port to kubernetes.